File: objectModel/Python/tests/cdm/projection/type_attribute_param.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

class TypeAttributeParam:
    """Type Attribute Test Parameters"""

    def __init__(self, name: str, data_type: str, purpose: str):
        self._attribute_name = name  # type: str
        self._attribute_data_type = data_type  # type: str
        self._attribute_purpose = purpose  # type: str

    # Attribute name property
    def get_attribute_name(self) -> str:
        return self._attribute_name

    def set_attribute_name(self, name: str) -> None:
        self._attribute_name = name

    # Attribute datatype property
    def get_attribute_data_type(self) -> str:
        return self._attribute_data_type

    def set_attribute_data_type(self, data_type: str) -> None:
        self._attribute_data_type = data_type

    # Attribute purpose property
    def get_attribute_purpose(self) -> str:
        return self._attribute_purpose

    def set_attribute_purpose(self, purpose: str) -> None:
        self._attribute_purpose = purpose

File: objectModel/Java/objectmodel/src/test/java/com/microsoft/commondatamodel/objectmodel/cdm/projection/TypeAttributeParam.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.cdm.projection;

import org.testng.annotations.Test;

/**
 * Type Attribute Test Parameters
 */
public class TypeAttributeParam {
    /**
     * Attribute name property
     */
    private String attributeName = "";

    public String getAttributeName() {
        return attributeName;
    }

    public void setAttributeName(String attributeName) {
        this.attributeName = attributeName;
    }

    /**
     * Attribute data type property
     */
    private String attributeDataType = "";

    public String getAttributeDataType() {
        return attributeDataType;
    }

    public void setAttributeDataType(String attributeDataType) {
        this.attributeDataType = attributeDataType;
    }

    /**
     * Attribute purpose property
     */
    private String attributePurpose = "";

    public String getAttributePurpose() {
        return attributePurpose;
    }

    public void setAttributePurpose(String attributePurpose) {
        this.attributePurpose = attributePurpose;
    }

    public TypeAttributeParam() {
        this(null, null, null);
    }

    public TypeAttributeParam(String name, String dataType, String purpose) {
        setAttributeName(name);
        setAttributeDataType(dataType);
        setAttributePurpose(purpose);
    }
}
